### PR TITLE
Retrieve and store license agreement

### PR DESCRIPTION
**Pull request by maartenba**

### CODE REVIEW

File: src/Catalog/Ownership/StorageRegistration.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System;
```
Comment (johnataylor) [inline src/Catalog/Ownership/StorageRegistration.cs]: I think we should just drop the whole StorageRegistration class at this point. We have the interface and the table implementation; its tedious keeping this up to date.

File: tests/PowerShellPublishTests/MockIRegistrationOwnership.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog.Ownership;
 using NuGet.Services.Publish;
 
 namespace PowerShellPublishTests
```
Comment (johnataylor) [inline tests/PowerShellPublishTests/MockIRegistrationOwnership.cs]: We can also drop the PowerShell specific code too. The PowerShell packages are just regular NuGet packages with a different type to filter on in search.

**Comment by johnataylor** [comment]
Looks great. Couple of comments on dropping a couple of files (and we could do that as a separate check in if you like.)

**Comment by maartenba** [comment]
Let's do those separately. Thanks @johnataylor

**Pull request merged**